x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/pagerduty.tsx

```diff
@@ -181,6 +186,56 @@ const PagerDutyParamsFields: React.FunctionComponent<ActionParamsProps<PagerDuty
     { value: 'resolve', text: 'Resolve' },
     { value: 'acknowledge', text: 'Acknowledge' },
   ];
+  const [isVariablesPopoverOpen, setIsVariablesPopoverOpen] = useState<Record<string, boolean>>({
+    dedupKey: false,
+    summary: false,
+    source: false,
+    timestamp: false,
+    component: false,
+    group: false,
+    class: false,
+  });
+  // TODO: replace this button with a proper Eui component, when it will be ready
+  const getMessageVariables = (paramsProperty: string) =>
+    messageVariables?.map((variable: string) => (
+      <EuiContextMenuItem
+        key={variable}
+        icon="empty"
+        onClick={() => {
+          editAction(
+            paramsProperty,
+            ((actionParams as any)[paramsProperty] ?? '').concat(` {{${variable}}}`),
+            index
+          );
+          setIsVariablesPopoverOpen({ ...isVariablesPopoverOpen, [paramsProperty]: false });
+        }}
+      >
+        {`{{${variable}}}`}
+      </EuiContextMenuItem>
+    ));
+
+  const getAddVariableComponent = (paramsProperty: string, buttonName: string) => {
+    return (
+      <EuiPopover
+        button={
+          <EuiButtonIcon
+            data-test-subj={`${buttonName}AddVariableButton`}
+            onClick={() => setIsVariablesPopoverOpen({ ...isVariablesPopoverOpen, dedupKey: true })}
```

Comment:
I'm noticing all the popovers fill values for dedup key, is it because `dedupKey: true` is hardcoded in this function?